filewatcher/filewatcher.go

```diff
@@ -135,14 +187,28 @@ func (r *Result) watcherLoop(
 			forceReload(mtime)
 		}
 	}
+	refreshWatched := func() {
+		for path := range watched {
+			watcher.Remove(path)
+			delete(watched, path)
+		}
+		err := filepath.WalkDir(path, func(p string, info fs.DirEntry, err error) error {
+			watched[p] = true
+			return watcher.Add(p)
```

Comment:
we should ignore the return of `watcher.Add` and always return `true` to the walkdir callback. the reason is for whatever reason fsnotify failed to watch a particular file/dir, that should not stop us from adding the rest of files/dirs to fsnotify.

Reply:
also we need to always add the parent directory of `path` additionally (likely before the `filepath.WalkDir` call), like we did in the initial parse.